libcxx/include/__iterator/iterator_traits.h

```diff
@@ -466,6 +467,18 @@ using __has_exactly_bidirectional_iterator_category _LIBCPP_NODEBUG =
 template <class _InputIterator>
 using __iter_value_type _LIBCPP_NODEBUG = typename iterator_traits<_InputIterator>::value_type;
 
+#if _LIBCPP_STD_VER >= 23
+template <class _InputIterator>
+using __iter_key_type _LIBCPP_NODEBUG = __remove_const_t<tuple_element_t<0, __iter_value_type<_InputIterator>>>;
```

Comment:
```suggestion
template <class _InputIterator>
using __iter_key_type _LIBCPP_NODEBUG = remove_const_t<tuple_element_t<0, __iter_value_type<_InputIterator>>>;
```
Did you make sure these are only used in the places the standard uses the exposition-only alias?

Reply:
Yes, I've verified that the uses of `__iter_key_type`, `__iter_mapped_type`, and `__iter_to_alloc_type` in libc++ exactly match the uses of _`iter-key-type`_, _`iter-mapped-type`_, and _`iter-to-alloc-type`_ in the standard wording.

Reply:
Thanks, @philnik777! Good point. For consistency with the original definition (which also used `__remove_const_t` - see https://github.com/llvm/llvm-project/blob/0ed1c9862df408d3df171762bd8fa0dafbe1b7e2/libcxx/include/__iterator/iterator_traits.h#L469-L471), and since @frederick-vs-ja verified the usage context of these internal aliases is correct per the standard, I'd prefer to keep it as is.

Reply:
I am aware that the original used the private version. That's purely due to necessity though. Not using the private version when we don't have to avoids cargo-culting.

Reply:
I see your point. Yes, the private version could be replaced with `std::remove_const_t` for 14 and later. Thanks, I'll make the change.